Sources/KIF/Additions/UIView-KIFAdditions.m

```diff
@@ -373,36 +376,26 @@ - (UIAccessibilityElement *)accessibilityElementMatchingBlock:(BOOL(^)(UIAccessi
                     }
 
                     @autoreleasepool {
-                        // Get the cell directly from the dataSource because UICollectionView will only vend visible cells
-                        UICollectionViewCell *cell = [collectionView.dataSource collectionView:collectionView cellForItemAtIndexPath:indexPath];
-
-                        // The cell contents might change just prior to being displayed, so we simulate the cell appearing onscreen
-                        if ([collectionView.delegate respondsToSelector:@selector(collectionView:willDisplayCell:forItemAtIndexPath:)]) {
-                            [collectionView.delegate collectionView:collectionView willDisplayCell:cell forItemAtIndexPath:indexPath];
-                        }
-                        
+                        CGRect visibleRect = [collectionView layoutAttributesForItemAtIndexPath:indexPath].frame;
+                        [collectionView scrollRectToVisible:visibleRect animated:NO];
+                        [collectionView layoutIfNeeded];
+                        UICollectionViewCell *cell = [collectionView cellForItemAtIndexPath:indexPath];
                         UIAccessibilityElement *element = [cell accessibilityElementMatchingBlock:matchBlock notHidden:NO disableScroll:NO];
-
-                        // Remove the cell from the collection view so that it doesn't stick around
-                        [cell removeFromSuperview];
                         
                         // Skip this cell if it isn't the one we're looking for
-                        // Sometimes we get cells with no size here which can cause an endless loop, so we ignore those
-                        if (!element || CGSizeEqualToSize(cell.frame.size, CGSizeZero)) {
+                        if (!element) {
                             continue;
                         }
                     }
                     
-                    // Scroll to the cell and wait for the animation to complete
-                    CGRect frame = [collectionView.collectionViewLayout layoutAttributesForItemAtIndexPath:indexPath].frame;
-                    [collectionView scrollRectToVisible:frame animated:YES];
                     // Note: using KIFRunLoopRunInModeRelativeToAnimationSpeed here may cause tests to stall
-                    CFRunLoopRunInMode(UIApplicationCurrentRunMode, 0.5, false);
-                    
-                    // Now try finding the element again
+                    CFRunLoopRunInMode(UIApplicationCurrentRunMode, CELL_SCROLL_DELAY_STABILIZATION, false);
                     return [self accessibilityElementMatchingBlock:matchBlock disableScroll:NO];
                 }
             }
+
+            [collectionView setContentOffset:initialPosition.origin];
```

Comment:
We do a `layoutIfNeeded` after `layoutIfNeeded` above; should we do that here too?

Reply:
Sure, doesn't really hurt. I'll add it for consistency.